### PR TITLE
cli: match version from build environment

### LINKED ISSUE #121: Use the MI version as the CLI version
Currently, the version used for MI CLI is different from the product version. Shall we use the same version for CLI too since it is easier to correlate?

We need to make sure the CLI version upgrade also happens whenever we do an MI release.

### PR DESCRIPTION
**Pull request by menuka94**

### CODE REVIEW

**Comment by Nirothipan** [comment]
@menuka94  this is intend to fix https://github.com/wso2/micro-integrator/issues/121 , isn't it

**Comment by menuka94** [comment]
> @menuka94 this is intend to fix #121 , isn't it

Yes. I'll update the commit message.

**Comment by a5anka** [comment]
As discussed offline, shall we try to read the version from pom.file for local builds?

**Comment by menuka94** [comment]
> As discussed offline, shall we try to read the version from pom.file for local builds?

Done.

File: Makefile [hunk 1]
```diff
@@ -14,11 +14,32 @@
...
+ifdef POM_VERSION
+VERSION=$(POM_VERSION)
+endif
+endif
+# one of these will be set to VERSION in Jenkins Production Environment
+
+
+ifndef VERSION
+# local build, use the POM version
+VERSION:=$(shell mvn -q -Dexec.executable=echo -Dexec.args='$${project.version}' --non-recursive exec:exec)
```
Comment (a5anka) [inline Makefile]: Why do we have two `$` characters?

**Comment by a5anka** [comment]
@menuka94 Do we need to read from the Jenkins environment now?

File: Makefile (hunk 1 above)
Comment (menuka94) [reply Makefile]: Because the expression inside parentheses needs to get evaluated and assigned to variable `VERSION`.

**Comment by menuka94** [comment]
> @menuka94 Do we need to read from the Jenkins environment now?

Yes, because otherwise the version will not match the Maven release version.

**Comment by a5anka** [comment]
> > @menuka94 Do we need to read from the Jenkins environment now?
> 
> Yes, because otherwise, the version will not match the Maven release version.

Then shall we have two make targets? One to use in Jenkins and one to use when building locally.

**Comment by menuka94** [comment]
> > > @menuka94 Do we need to read from the Jenkins environment now?
> > 
> > Yes, because otherwise, the version will not match the Maven release version.
> 
> Then shall we have two make targets? One to use in Jenkins and one to use when building locally.

Done.

**Comment by a5anka** [comment]
@menuka94 Can you let me know when we can merge this without breaking the product build?

**Comment by menuka94** [comment]
> @menuka94 Can you let me know when we can merge this without breaking the product build?

Sure. Even if this is merged before deploying the new Maven Release Plugin in Jenkins, general builds will work fine. The only caveat is that in release builds, the CLI will not have the new release version.

**Comment by menuka94** [comment]
@a5anka This can be merged now.

**Pull request merged**